execute/plugin.go

```diff
@@ -620,23 +626,36 @@ func (p *Plugin) checkAlreadyExecuted(
 ) error {
 	seqNrRangesBySource := getSeqNrRangesBySource(reports)
 
-	// TODO: batch these queries? these are all DB reads.
-	// maybe some alternative queries exist.
-	for sourceChainSelector, seqNrRange := range seqNrRangesBySource {
-		executed, err := p.ccipReader.ExecutedMessages(ctx, sourceChainSelector, seqNrRange, primitives.Unconfirmed)
-		if err != nil {
-			return fmt.Errorf("couldn't check if messages already executed: %w", err)
+	executed, err := p.ccipReader.ExecutedMessages(
+		ctx,
+		seqNrRangesBySource,
+		primitives.Unconfirmed)
+	if err != nil {
+		return fmt.Errorf("couldn't check if messages already executed: %w", err)
+	}
+
+	for sourceChainSelector, expectedSqNrRanges := range seqNrRangesBySource {
+		snRangeSet := mapset.NewSet[cciptypes.SeqNum]()
```

Comment:
```suggestion
		snSet := mapset.NewSet[cciptypes.SeqNum]()
```

This is a set of seq nrs not ranges